### PR TITLE
Repobuild improvements for Ubuntu 18.04 lack of gpg2 and better error checking

### PR DESCRIPTION
**Pull request by dmurphy18**
### What does this PR do?
Handles Ubuntu 18.04 removal of gpg2, now named gpg.  Took opportunity to improve error checking.

### What issues does this PR fix or reference?
Ubuntu 18.04 removed gpg2 and gpg1 from default installation, now has plain gpg.
The changes in debbuild.py compensate for this change while still allow ability to build on Ubuntu 14.04 and 16.04.

### New Behavior
Better error checking for shelled out commands used to build
Compensates for lack of gpg2 on Ubuntu 18.04

### Tests written?
No, hand built and signed packages on Ubuntu 18.04 for Python 2, and built packages for Python 3.

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/debbuild.py [hunk 1]
```diff
@@ -600,7 +610,7 @@ def make_repo(repodir,
     # preset passphase and interaction with gpg-agent
     gpg_info_file = '{0}/gpg-agent-info-salt'.format(gnupghome)
     gpg_tty_info_file = '{0}/gpg-tty-info-salt'.format(gnupghome)
-    gpg_tty_info_dict = {}
+#    gpg_tty_info_dict = {}
```
Comment (cachedout) [inline salt/modules/debbuild.py]: If this isn't used, let's just remove it entirely.

File: salt/modules/debbuild.py [hunk 2]
```diff
@@ -571,12 +575,18 @@ def make_repo(repodir,
...
-            'stderr': 'initialization value'}
+            'stderr': 'initialization value'
+            }
+
+    retrc = 0
 
-    SIGN_PROMPT_RE = re.compile(r'Enter passphrase: ', re.M)
-    REPREPRO_SIGN_PROMPT_RE = re.compile(r'Passphrase: ', re.M)
+    if gnupghome:
+        if env is None:
```
Comment (cachedout) [inline salt/modules/debbuild.py]: This might be cleaner if we AND the previous conditional.

File: salt/modules/debbuild.py [hunk 3]
```diff
@@ -755,19 +778,23 @@ def make_repo(repodir,
                 number_retries = timeout / interval
                 times_looped = 0
                 error_msg = 'Failed to reprepro includedsc file {0}'.format(abs_file)
-                cmd = 'reprepro --ignore=wrongdistribution --component=main -Vb . includedsc {0} {1}'.format(codename, abs_file)
+                cmd = 'reprepro --ignore=wrongdistribution --component=main -Vb . includedsc {0} {1}'.format(
+                        codename,
+                        abs_file)
+
+#               env=gpg_tty_info_dict,
```
Comment (cachedout) [inline salt/modules/debbuild.py]: Let's also please remove this commented line.

File: salt/modules/debbuild.py (hunk 2 above)
Comment (dmurphy18) [reply salt/modules/debbuild.py]: sure, had thought about doing that but wanted to match code in PR https://github.com/saltstack/salt/pull/47056, and clean it up later, but can do it now

File: salt/modules/debbuild.py (hunk 1 above)
Comment (dmurphy18) [reply salt/modules/debbuild.py]: Sorry about that, usually have my commented out with a DGM prefix, and in this case didn't, hence missed in search for cleanup,, usage cleaned up.

File: salt/modules/debbuild.py (hunk 3 above)
Comment (dmurphy18) [reply salt/modules/debbuild.py]: done

**Review by cachedout — APPROVED** [review]

**Pull request merged**